gcloudutils/cloudbuild_test.go

```diff
@@ -0,0 +1,191 @@
+)
+
+var (
+	bfs  cloudbuild.Build
+	bfr  cloudbuild.Build
+	bfrt cloudbuild.Build
+)
+
+const (
+	devDeploy = "dev-deploy"
+	tag_name  = "v0.1.0"
+)
+
+var _ = BeforeSuite(func() {
+	var err error
+	err = json.Unmarshal([]byte(buildFromSource), &bfs)
+	Expect(err).NotTo(HaveOccurred())
+	err = json.Unmarshal([]byte(buildFromRepo), &bfr)
+	Expect(err).NotTo(HaveOccurred())
+	err = json.Unmarshal([]byte(buildFromRepoTag), &bfrt)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = Describe("github test", func() {
+
+	Context("repo name", func() {
+		It("can find the repo name from repo source", func() {
+			Expect(gcloudutils.GetRepoName(&bfr)).To(Equal(devDeploy))
+		})
+	})
+
+	Context("tag name", func() {
+
+		It("can find tag name from source", func() {
+			tag := gcloudutils.GetReleaseVersionTag(&bfrt)
+			Expect(tag).To(Equal(tag_name))
+		})
+
+		It("returns empty string if no tag exists", func() {
+			tag := gcloudutils.GetReleaseVersionTag(&bfr)
+			Expect(tag).To(Equal(""))
+		})
+	})
+
+	Context("tags", func() {
+		It("can find release tag from tags", func() {
+			tag := gcloudutils.GetReleaseVersionTag(&bfs)
+			Expect(tag).To(Equal(tag_name))
+		})
+
+		It("can find sha from tags", func() {
+			tag := gcloudutils.GetTargetCommitSh(&bfs)
+			Expect(tag).To(Equal("world"))
+		})
+
+		It("can find repo name from tags", func() {
+			tag := gcloudutils.GetRepoName(&bfs)
+			Expect(tag).To(Equal(devDeploy))
+
+		})
+	})
+})
+
+const buildFromSource = `
+    {
+      "id": "6c114a53-872a-4b3b-aabb-ea547c9e9406",
+      "status": "SUCCESS",
+      "source": {
+        "storageSource": {
+          "bucket": "solo-corp_cloudbuild",
+          "object": "source/solobot/1550549389.83-14f20ad743af4b7c942726aadfd12c33.tgz",
+          "generation": "1550549390047482"
+        }
+      },
+      "tags": [
+      	"ref_hello",
+      	"tag_v0.1.0",
+      	"sha_world",
+      	"repo_dev-deploy"
+      ],
+      "createTime": "2019-02-19T04:09:50.309127374Z",
+      "startTime": "2019-02-19T04:09:51.269815120Z",
+      "finishTime": "2019-02-19T04:13:10.228509Z",
+      "results": {},
+      "steps": [],
+      "timeout": "600s",
+      "projectId": "solo-corp",
+      "logsBucket": "gs://726673624965.cloudbuild-logs.googleusercontent.com",
+      "sourceProvenance": {
+        "resolvedStorageSource": {
+          "bucket": "solo-corp_cloudbuild",
+          "object": "source/solobot/1550549389.83-14f20ad743af4b7c942726aadfd12c33.tgz",
+          "generation": "1550549390047482"
+        },
+        "fileHashes": {
+          "gs://solo-corp_cloudbuild/source/solobot/1550549389.83-14f20ad743af4b7c942726aadfd12c33.tgz#1550549390047482": {}
+        }
+      },
+      "options": {
+        "logging": "LEGACY"
+      },
+      "substitutions": {
+        "TAG_NAME": "v0.1.0"
+      },
+      "logUrl": "https://console.cloud.google.com/gcr/builds/6c114a53-872a-4b3b-aabb-ea547c9e9406?project=726673624965"
+    }
+`
+
+const buildFromRepo = `
```

Comment:
FWIW any tests using this are no longer necessary as we don't use RepoSource for our builds anymore